src/backend/app/projects/project_crud.py

```diff
@@ -282,3 +286,42 @@ async def preview_split_by_square(boundary: str, meters: int):
             meters=meters,
         )
     )
+
+
+def process_imagery_in_nodeodm(images_folder: str, project_name: str):
+    token = ""
+    task_name = project_name
+    images_dir = images_folder
+    node_odm_docker_url = settings.NODE_ODM_INTERNAL_DOCKER_URL
+    task_processor = TaskProcessor(token, task_name, images_dir, node_odm_docker_url)
+    task_processor.process()
+
+
+def download_object(bucket_name: str, obj, images_folder: str):
+    if obj.object_name.endswith((".jpg", ".jpeg", ".JPG", ".png", ".PNG")):
+        log.info(f"Downloading image from s3 {obj.object_name}")
+        local_path = f"{images_folder}/{os.path.basename(obj.object_name)}"
+        os.makedirs(os.path.dirname(local_path), exist_ok=True)
+        get_file_from_bucket(bucket_name, obj.object_name, local_path)
+
+
+def process_images_concurrently(project_id: uuid.UUID, task_id: uuid.UUID):
+    objects = list_objects_from_bucket(
+        settings.S3_BUCKET_NAME, f"publicuploads/{project_id}"
+    )
+
+    # Construct the images folder path
+    images_folder = (
+        f"/tmp/{project_id}/{task_id}"  # NOTE: local path is set to /tmp/ folder.
+    )
+    project_name = "Testing Project"  # FIXME: Use real Project name here
+
+    # Process images concurrently
+    with ThreadPoolExecutor() as executor:
```

Comment:
I wouldn't recommend usage of the threadpool in an async framework like FastAPI.

This approach will work and could be fine for now, but it might choke the server under load as threads are overallocated.

It's much more resource efficient to process this asynchronously using an aiohttp connection pool, but it does mean installing something like aiohttp as a dependency.

Downloading multiple files this way only uses one thread. The downloads are added to a queue and the selector checks when the download is ready to be processed by the OS and opens a connection as needed. It's more efficient than having multiple threads open connections and keeping them open until done